### PR TITLE
oath-toolkit: replace update script with nix-update

### PR DESCRIPTION
**Pull request by schnusch**
nix-update supports GNU Savannah
see https://github.com/Mic92/nix-update/releases/tag/0.5.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
But does nixpkgs-update? :thinking:

**Comment by schnusch** [comment]
Good question. I thought nixpkgs-update uses nix-update under the hood, but I cannot verify that, because I cannot get it to run myself.
```nixpkgs-update: Maybe.fromJust: Nothing
CallStack (from HasCallStack):
  error, called at libraries/base/Data/Maybe.hs:148:21 in base:Data.Maybe
  fromJust, called at src/Utils.hs:306:12 in nixpkgs-update-0.4.0-1LZpx5N2l6q8e22qNSkACg:Utils```

**Comment by mweinelt** [comment]
@r-ryantm runs nixpkgs-update (haskell), not nix-update (python).

File: pkgs/tools/security/oath-toolkit/default.nix [hunk 1]
```diff
@@ -19,8 +19,6 @@ in stdenv.mkDerivation rec {
 
   configureFlags = lib.optionals stdenv.isDarwin [ "--disable-pam" ];
 
-  passthru.updateScript = ./update.sh;
```
Comment (winterqt) [inline pkgs/tools/security/oath-toolkit/default.nix:22]: To let `nixpkgs-update` update the package, do this:
```nix
passthru.updateScript = nix-update-script { attrPath = pname; };
```

Which, as the name gives away, uses `nix-update` :)

**Review by winterqt — CHANGES_REQUESTED** [review]

**Comment by winterqt** [comment]
I'm torn on if a message like `use nix-update` or something like that would be more clear. The diff is so small though that it probably doesn't matter, so I guess I'm fine with either.

What do you think, am I looking too much into it? :-)

**Review by winterqt — APPROVED** [review]
LGTM, though.

**Comment by mweinelt** [comment]
Remove is clearly the wrong word now. Replace would be better.

**Comment by winterqt** [comment]
@mweinelt What do you think of the one just pushed, or would `replace update script with nix-update` be better? I think either work, not sure if any particular one is better, though.

**Comment by mweinelt** [comment]
Ideally it wouldn't say `Remove update script`, because that could be interpreted as removing `passthru.updateScript`.

**Comment by winterqt** [comment]
Yeah, that was also my concern, but I didn't want to be too nitpicky.

**Pull request merged**